### PR TITLE
Gltf Refactor

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

As mentioned in #4597, this refactors the GLTF functionality **while preserving the existing output**

This is an absolute minimal fix to have the tests passing, but I think we should refactor:
* Remove all static storage objects
* change the data layout and API in the classes that are in GltfForwardTranslator
    * just use `T` (eg double) instead of std::tuple<std::string, tinygltf::Value(T)> 
    * provide a protected toGltfValue that will create the std::map<std::string, tinygltf::Value>
* Supporting classes:
    * Move those to separate files (like the wrappers are)
    * Remove the wrappers (which are superflous now)
    * What we now have are **Interface** classes that can go to/from ModelObject and GLTF
        *  There is (at least) one Ctor that takes in Model or a specific derived ModelObject to populate the interface. Later you can call to Extras() (or similar) to convert that to a tinygltf object
        * There is a ctor that takes in a tinygltf::Value (or tinygltf::Material) to populate the class from the other way
        * **Note:** what's missing to produce a GltfReverseTranslator in the ability to output ModelObjects from the Interface class
* Moved everything into the gltf folder & gltf namespace for clarity (eventually we'll probably have a proper GltfReverseTranslator too...)

* I have exposed an modelToGltfString method for ease of consumption/checking, and in SWIG I provide modelToGltfJSON helper that returns a ruby hash directly.

* Wrote some docstring in the classes and a mainpage (TODO: need to check that Doxygen renders it correctly. **Edit:**, Build it locally, works fine)

* Started JSON schemas for the `extras` we place on the GLTF Nodes and the GLTF Scene
    * For documentation purposes, but later we could do validation too. 

* I have left a bunch of `TODOs` in the code (it's in its own folder, so easy to grep for it...), and will document some potential next steps/additional refactor in a separate issue #4606

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/gltf/GltfModelObjectMetaData.hpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    /** @name Interface with ModelObjects */
+    //@{
+    explicit GltfModelObjectMetaData(const model::AirLoopHVAC& airLoopHVAC);
+    explicit GltfModelObjectMetaData(const model::BuildingUnit& buildingUnit);
+    explicit GltfModelObjectMetaData(const model::SpaceType& spaceType);
+    explicit GltfModelObjectMetaData(const model::BuildingStory& buildingStory);
+    explicit GltfModelObjectMetaData(const model::ThermalZone& thermalZone);
+    explicit GltfModelObjectMetaData(const model::DefaultConstructionSet& defaultConstructionSet);
+    explicit GltfModelObjectMetaData(const model::Space& space);
+    //@}
```
Comment (jmarrec) [inline src/gltf/GltfModelObjectMetaData.hpp:76]: I won't be commenting every class, but I'll do this one as an example.

As you can see, this has several constructor that take in a derived ModelObject to populate the interface members

File: src/gltf/GltfModelObjectMetaData.hpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+   protected:
+    /** @name Protected */
+    //@{
+
+    /** Interface with an existing GLTF file **/
+    explicit GltfModelObjectMetaData(const tinygltf::Value& modelObjectMetaDataObj);
+
+    /** Export to tinygltf **/
+    std::map<std::string, tinygltf::Value> toExtras() const;
```
Comment (jmarrec) [inline src/gltf/GltfModelObjectMetaData.hpp:115]: Protected area: 

The Ctor  is to do the reverse: populate from a GLTF object

The toExtras exports to a tinygltf object

**Comment by jmarrec** [comment]
ok CI is happy @tijcolem

File: CMakeLists.txt [hunk 3]
```diff
@@ -1007,6 +1007,7 @@ set(project_directories
   epjson
   radiance
   gbxml
+  gltf
```
[the remark this answers is not in the document]
Comment (tijcolem) [reply CMakeLists.txt:1010]: yep, makes sense

File: src/gltf/schemas/node.extras.schema.json [hunk 4]
```diff
@@ -0,0 +1,210 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
```
Comment (tijcolem) [inline src/gltf/schemas/node.extras.schema.json:2]: great adding the schema!

File: src/gltf/schemas/scene.extras.schema.json [hunk 5]
```diff
@@ -0,0 +1,183 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "scene.extras.schema.json",
```
Comment (tijcolem) [inline src/gltf/schemas/scene.extras.schema.json:3]: same here. great!

**Review by tijcolem — APPROVED** [review]

**Pull request merged**